jobclient/python/jobs.py

```diff
@@ -0,0 +1,381 @@
+# Copyright (c) Two Sigma Open Source, LLC
+#
+# Licensed under the Apache license, Version 2.0 (the "License");
+# you may not use this file ecept in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+import constraints
+import util
+
+from copy import deepcopy
+from enum import Enum
+from uuid import UUID
+from typing import Dict, List, Optional, Set
+
+from constraints import Constraint
+from instance import Executor, Instance
+from util import FetchableUri
+
+
+class Status(Enum):
+    """
+    The curent status of a job.
+
+    A job in Cook has only three statuses: WAITING, RUNNING, COMPLETED. The
+    INITIALIZED status is for a `JobListener` to receive status updates.
```

Comment:
I think we should not do the `JobListener` stuff in this PR.

Reply:
:+1: